oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtAuthenticationConverter.java

```diff
@@ -43,8 +44,9 @@ public final AbstractAuthenticationToken convert(Jwt jwt) {
 		if (this.principalClaimName == null) {
 			return new JwtAuthenticationToken(jwt, authorities);
 		}
-		String name = jwt.getClaim(this.principalClaimName);
-		return new JwtAuthenticationToken(jwt, authorities, name);
+		Object principalClaimValue = jwt.getClaim(this.principalClaimName);
```

Comment:
Would it be better to call `jwt.getClaimAsString(this.principalClaimName)`? The benefit is that the converter would work for principals of other types. I think this aligns with how `AbstractAuthenticationToken#getName` also coerces the principal into a string.

cc @jgrandja

Reply:
@jzheaux I initially submit this issue to avoid a ClassCastException and I suggested both ideas to achieve this.
We went for the other one.

After this, I agree with you that we should keep the alignment with JwtClaimAccessor#getSubject method by using jwt.getClaimAsString(this.principalClaimName). It will also reduce the situation where we can have a ClassCastException.

I can change the code, just let me know what to do 👍

Reply:
Makes sense, @omlip.

Yes, let's go ahead and make the change to `getClaimAsString`. This is also consistent with `DefaultOAuth2User#getName` which coerces the configured claim into a `String`.

Reply:
ok thanks, will adapt soon.